x-pack/plugins/observability_solution/observability_ai_assistant/server/plugin.ts

```diff
@@ -107,22 +107,24 @@ export class ObservabilityAIAssistantPlugin
       };
     }) as ObservabilityAIAssistantRouteHandlerResources['plugins'];
 
+    // Using once to make sure the same model ID is used during service init and Knowledge base setup
     const getModelId = once(async () => {
-      // Using once to make sure the same model ID is used during service init and Knowledge base setup
+      const defaultModelId = '.elser_model_2';
+      const [_, pluginsStart] = await core.getStartServices();
+      const license = await firstValueFrom(pluginsStart.licensing.license$);
+
+      if (!license.hasAtLeast('enterprise')) {
```

Comment:
Platinum?

Reply:
I was not sure. We require `enterprise` here:
https://github.com/elastic/kibana/blob/12ecb70e185680e9c8e0a333c5f381904a2000b3/x-pack/plugins/observability_solution/observability_ai_assistant_app/server/rule_connector/index.ts#L96

Reply:
yes, but I assume this throws because users don't have a platinum license (for ML). maybe we should guard elsewhere 🤔 maybe in `/server/service/index.ts:doInit` we should catch the error, WDYT?

Reply:
Yes, it's because of ML but regardless we should never attempt to do this if the customer does not have enterprise (which is the required license for AI Assistant.

Reply:
> maybe we should guard elsewhere 🤔 maybe in /server/service/index.ts:doInit we should catch the error, WDYT?

What about `setup` then? 

https://github.com/elastic/kibana/blob/cff6de2ecf47a0e9068b8df0b8ae044ae602005d/x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts#L80-L81

or status?

https://github.com/elastic/kibana/blob/cff6de2ecf47a0e9068b8df0b8ae044ae602005d/x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts#L273-L274